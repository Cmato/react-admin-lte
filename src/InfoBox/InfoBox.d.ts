import * as React from 'react';

export interface InfoBoxProps {
  backgroundColor: 'aqua' | 'green' | 'yellow' | 'red';
  backgroundPlacement: InfoBoxBackgroundPlacement;
  icon: React.ReactNode;
  number: React.ReactNode;
  text: React.ReactNode;
}

export type InfoBoxBackgroundPlacement = 'box' | 'icon';

declare const InfoBox: React.ComponentType<InfoBoxProps>;

export default InfoBox;
