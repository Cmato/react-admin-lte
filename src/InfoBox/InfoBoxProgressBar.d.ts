import * as React from 'react';
import { ProgressBarProps } from 'react-bootstrap/lib/ProgressBar';

export interface InfoBoxProgressBarProps extends ProgressBarProps {
  description: React.ReactNode;
}

declare const InfoBoxProgressBar: React.ComponentType<InfoBoxProgressBarProps>;

export default InfoBoxProgressBar;
