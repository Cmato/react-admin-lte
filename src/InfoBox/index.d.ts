export { default } from './InfoBox';
export * from './InfoBox';
export { default as InfoBoxProgressBar } from './InfoBoxProgressBar';
export * from './InfoBoxProgressBar';
