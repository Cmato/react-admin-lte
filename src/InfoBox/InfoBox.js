import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const InfoBox = ({
  backgroundColor,
  backgroundPlacement,
  children,
  className: classNameProp,
  icon,
  number,
  progress,
  progressDescription,
  text,
  ...other
}) => {
  const boxClassName = classNames({
    'info-box': true,
    [`bg-${backgroundColor}`]: backgroundPlacement === 'box'
  }, classNameProp);

  const iconClassName = classNames({
    'info-box-icon': true,
    [`bg-${backgroundColor}`]: backgroundPlacement === 'icon'
  });

  return (
    <div className={boxClassName} {...other}>
      <span className={iconClassName}>{icon}</span>
      <div className="info-box-content">
        <span className="info-box-text">{text}</span>
        <span className="info-box-number">{number}</span>
        {children}
      </div>
    </div>
  );
};

InfoBox.propTypes = {
  /**
   * Color of background.
   */
  backgroundColor: PropTypes.oneOf(['aqua', 'green', 'yellow',  'red']).isRequired,
  /**
   * Whether the `backgroundColor` should be applied to only the `icon` or the whole `box`.
   */
  backgroundPlacement: PropTypes.oneOf(['box', 'icon']).isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * An icon to be displayed on the left side of the `InfoBox`.
   */
  icon: PropTypes.node.isRequired,
  /**
   * The info to be displayed in `InfoBox`.
   */
  number: PropTypes.node.isRequired,
  /**
   * The title to be displayed in `InfoBox`.
   */
  text: PropTypes.node.isRequired
};

export default InfoBox;
