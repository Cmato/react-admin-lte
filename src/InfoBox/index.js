export { default } from './InfoBox';
export { default as InfoBoxProgressBar } from './InfoBoxProgressBar';
