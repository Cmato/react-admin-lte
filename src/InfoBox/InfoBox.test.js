import React from 'react';
import { shallow } from 'enzyme';
import InfoBox from './InfoBox';

it('renders icon background', () => {
  expect(shallow((
    <InfoBox
      backgroundColor="red"
      backgroundPlacement="icon"
      icon={<div />}
      number={67}
      text="LIKES"
    >
      <div />
    </InfoBox>
  ))).toMatchSnapshot();
});

it('renders box background', () => {
  expect(shallow((
    <InfoBox
      backgroundColor="red"
      backgroundPlacement="box"
      icon={<div />}
      number={67}
      text="LIKES"
    >
      <div />
    </InfoBox>
  ))).toMatchSnapshot();
});

it('passes other props to root', () => {
  expect(shallow(
    <InfoBox
      backgroundColor="red"
      backgroundPlacement="box"
      icon={<div />}
      number={67}
      text="LIKES"
      other="thing"
    >
      <div />
    </InfoBox>
  )).toMatchSnapshot();
});
