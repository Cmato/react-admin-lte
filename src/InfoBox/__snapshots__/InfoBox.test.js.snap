// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes other props to root 1`] = `
<div
  className="info-box bg-red"
  other="thing"
>
  <span
    className="info-box-icon"
  >
    <div />
  </span>
  <div
    className="info-box-content"
  >
    <span
      className="info-box-text"
    >
      LIKES
    </span>
    <span
      className="info-box-number"
    >
      67
    </span>
    <div />
  </div>
</div>
`;

exports[`renders box background 1`] = `
<div
  className="info-box bg-red"
>
  <span
    className="info-box-icon"
  >
    <div />
  </span>
  <div
    className="info-box-content"
  >
    <span
      className="info-box-text"
    >
      LIKES
    </span>
    <span
      className="info-box-number"
    >
      67
    </span>
    <div />
  </div>
</div>
`;

exports[`renders icon background 1`] = `
<div
  className="info-box"
>
  <span
    className="info-box-icon bg-red"
  >
    <div />
  </span>
  <div
    className="info-box-content"
  >
    <span
      className="info-box-text"
    >
      LIKES
    </span>
    <span
      className="info-box-number"
    >
      67
    </span>
    <div />
  </div>
</div>
`;
